src/Hooks.ts

```diff
@@ -143,3 +145,75 @@ export function useFocusState() {
 
   return focusState;
 }
+
+type EffectCallback = (() => void) | (() => () => void);
+
+// Inspired by same hook from react-navigation v5
+// See https://github.com/react-navigation/hooks/issues/39#issuecomment-534694135
+export const useFocusEffect = (callback: EffectCallback) => {
+  const navigation = useNavigation();
+  const getCallback = useGetter(callback);
+
+  useEffect(() => {
+    let isFocused = false;
+    let cleanup: (() => void) | void;
+
+    if (navigation.isFocused()) {
+      cleanup = getCallback()();
+      isFocused = true;
+    }
+
+    const focusSubscription = navigation.addListener('willFocus', () => {
+      // If callback was already called for focus, avoid calling it again
+      // The focus event may also fire on intial render, so we guard against runing the effect twice
+      if (isFocused) {
+        return;
+      }
+
+      cleanup && cleanup();
+      cleanup = getCallback()();
+      isFocused = true;
+    });
+
+    const blurSubscription = navigation.addListener('willBlur', () => {
```

Comment:
Maybe we need to take `didFocus` and `didBlur` into account too? I'm not too sure which ones fire for other navigators such as tabs and drawer. Ignore if you're sure `willFocus` and `willBlur` fire universally.

Reply:
Hmmm will have to check. Tried to use the same events you have for v5 so that migration is easier.

Reply:
I think all events are emitted in a generic way so it should be safe to expect all navigators to fire the same will/did events. 

According to the comments, for navigators without transition, didFocus happen  immediately after willFocus.

https://github.com/react-navigation/core/blob/cdc4d9192d5d3add1739cde47034050e52df70b4/src/getChildEventSubscriber.js